ml-agents/mlagents/trainers/tests/torch/test_encoders.py

```diff
@@ -0,0 +1,109 @@
+import torch
+from unittest import mock
+import pytest
+
+from mlagents.trainers.torch.encoders import (
+    VectorEncoder,
+    VectorAndUnnormalizedInputEncoder,
+    Normalizer,
+    SimpleVisualEncoder,
+    ResNetVisualEncoder,
+    NatureVisualEncoder,
+)
+
+
+# This test will also reveal issues with states not being saved in the state_dict.
+def compare_models(module_1, module_2):
+    is_same = True
+    for key_item_1, key_item_2 in zip(
+        module_1.state_dict().items(), module_2.state_dict().items()
+    ):
+        is_same = torch.equal(key_item_1[1], key_item_2[1]) and is_same
```

Comment:
Why compare only the first item? `key_item_1[*1*]`?

Reply:
It's the ZIP, `key_item_X` is a Tuple of (key, value). So we're just comparing the values.